Sources/Observer.swift

```diff
@@ -68,6 +84,15 @@ extension Signal {
 			}
 		}
 
+		deinit {
+			if interruptsOnDeinit {
+				// Since `Signal` would ensure that only one terminal event would ever be
+				// sent for any given `Signal`, we do not need to assert any condition
+				// here.
+				action(.interrupted)
```

Comment:
Would it mess up anything if we always sent interrupted on `deinit`? I'm wondering if that'd be a better option.